File: SMI-SDK/.NET SDK/Library/gis/trace/HTTPMessageEntryProperty.cs
using System;
using System.Collections.Generic;
namespace pt.sapo.gis.trace
{

    /**
    * @author GIS Team 
     * Entry property to wrap an HTTP message
     */
    public class HTTPMessageEntryProperty : MessageEntryProperty
    {
        public static readonly String HEADERS_PROPERTY_NAME = "headers";
        public static readonly String BODY_PROPERTY_NAME = "body";

        public IDictionary<String, String> Headers
        {
            get { return (IDictionary<String, String>)this[HEADERS_PROPERTY_NAME]; }
            set { this[HEADERS_PROPERTY_NAME] = value; }
        }

        public String Body
        {
            get { return (String)this[BODY_PROPERTY_NAME]; }
            set { this.Add(BODY_PROPERTY_NAME, value); }
        }
    }
}

File: SMI-SDK/Java SDK/src/pt/sapo/gis/trace/HTTPMessageEntryProperty.java
package pt.sapo.gis.trace;

import java.util.HashMap;
import java.util.Map;

/**
* @author GIS Team 
 * Entry property to wrap an HTTP message
 */
public class HTTPMessageEntryProperty extends MessageEntryProperty {
    public static final String HEADERS_PROPERTY_NAME = "headers";
    public static final String BODY_PROPERTY_NAME = "body";
    
    public Map<String, String> getHeaders() {
        return (Map<String, String>) this.get(HEADERS_PROPERTY_NAME);
    }

    public void setHeaders(Map<String, String> headers) {
        this.put(HEADERS_PROPERTY_NAME, headers);
    }

    public void addHeader(String name, String value) {
        if(getHeaders() == null)
            setHeaders(new HashMap<String, String>());
        getHeaders().put(name, value);
    }
    
    public String getBody() {
        return (String) this.get(BODY_PROPERTY_NAME);
    }

    public void setBody(String body) {               
        this.put(BODY_PROPERTY_NAME, body);
    }    
}